src/stateProviders/provideEntity.ts

```diff
@@ -72,7 +73,29 @@ const provideEntity = provideState({
                     ...entity.associated[currentType],
                     ...Object.keys(change[currentType]).reduce((actions, action) => {
                       const otherAction = action === 'add' ? 'remove' : 'add';
+
                       if (
+                        findIndex(
+                          entity.associated[currentType][action],
+                          e => e.id && e.id === change[currentType][action].id,
+                        ) > -1
```

Comment:
`_.some` might read better :) there's also `Array.prototype.some`

Reply:
this one i should probably just save the result from `findIndex` since i need it if it is a valid index. it's for tracking the new staged associated changes (the ui lets you add multiple new permissions but you still need to modify the accessLevel)